### ISSUE TITLE
[FEATURE REQUEST] Fanart API adjustment 

### ISSUE THREAD

**Issue by starmanrf** [Ezra-Hubbard/Ezra-Hubbard.github.io#2]
Hey, solid job on keeping the project going...works like a charm! Kudos!!

I was wondering if you would be able to find time to add a line to pull keyart movie and TV show posters when available from the fanarttv API. I've tried having a go of it myself but haven't been able to figure it out 🙄.

Keep up the great work!

**Comment by Ezra-Hubbard** [comment]
will put it on the to-do list for "if i can figure it out" :D

**Comment by 123Venom** [comment]
keyart from FanartTV has a `lang` string values of `00` or `None`.  It can also be a None type.

```
try:
    keyart = art['movieposter']
    keyart = [(x['url'], x['likes']) for x in keyart if any(value == x.get('lang') for value in ('00', 'None', None))]
    if len(keyart) >1:keyart = sorted(keyart, key=lambda x: int(x[1]), reverse=True)
    keyart = [x[0] for x in keyart][0]
except: keyart = ''
```

**Comment by adamosborne83** [comment]
> keyart from FanartTV has a `lang` string values of `00` or `None`. It can also be a None type.
> 
> ```
> try:
>     keyart = art['movieposter']
>     keyart = [(x['url'], x['likes']) for x in keyart if any(value == x.get('lang') for value in ('00', 'None', None))]
>     if len(keyart) >1:keyart = sorted(keyart, key=lambda x: int(x[1]), reverse=True)
>     keyart = [x[0] for x in keyart][0]
> except: keyart = ''
> ```

Amazing to still see you around and commenting....

**Comment by umbrellaplug** [comment]
Added in Pull Request #2, Let me know if you have any issues.

**Comment by Ezra-Hubbard** [comment]
> Added in Pull Request #2, Let me know if you have any issues.

cheers @umbrellaplug

**Comment by starman2908** [comment]
Thank you guys so much...  I tried figuring out @123Venom code rand still couldn't get it working ... coding gods amongst men !

**Comment by 123Venom** [comment]
quick look at the fanarttv module it uses a method called `parse_art()` but that is designed to pull the matching artwork for the current language.  You might need to add a special `parse_keyart()` method or perhaps trick line 44 into failing.  If line 44 fails then line 45 uses the global `blank_image_values = ('00', '')`.  It's missing the `None` case I mentioned.

Even if you got the fanarttv module passing keyart you'd have to edit the artwork assignment
`listitem.setArt({'poster': poster, 'fanart': fanart, 'icon': poster, 'banner': banner, 'clearart': clearart,
    'clearlogo': clearlogo, 'landscape': landscape, 'discart': discart})`.  keyart is not set there anyway and would need to be in the list builder or the skin won't ever see it to use it.
    
fun fun.

**Comment by umbrellaplug** [comment]
> quick look at the fanarttv module it uses a method called `parse_art()` but that is designed to pull the matching artwork for the current language. You might need to add a special `parse_keyart()` method or perhaps trick line 44 into failing. If line 44 fails then line 45 uses the global `blank_image_values = ('00', '')`. It's missing the `None` case I mentioned.
> 
> Even if you got the fanarttv module passing keyart you'd have to edit the artwork assignment `listitem.setArt({'poster': poster, 'fanart': fanart, 'icon': poster, 'banner': banner, 'clearart': clearart, 'clearlogo': clearlogo, 'landscape': landscape, 'discart': discart})`. keyart is not set there anyway and would need to be in the list builder or the skin won't ever see it to use it.
> 
> fun fun.

I'm sure you get tired of reading it, but it's awesome to see you still around. I pushed a commit up to him earlier where it was getting the keyart, but I wasn't sure what exactly they were doing with it. From the brief research I did on it between things, it looks to just be like Movie Posters without text so the theme people can put the stuff in. I did a quick search on Fanart.tv and couldn't really find an example of any movie with it. I'll do a little more research on that side and find one and get him some more conditional checking in there for the none condition and the language check. I'll have to look at the list builder and place a conditional checking for keyart there for him too.... as you said.

fun fun.

**Comment by 123Venom** [comment]
Sorry I didn't see a PR.  The link you posted...
[#2](https://github.com/Ezra-Hubbard/Ezra-Hubbard.github.io/issues/2)
Just returns here to this issue and is not a Pull Request.  I then checked the PR's and open/closed was zilch
https://github.com/Ezra-Hubbard/Ezra-Hubbard.github.io/pulls

Not sure what your PR #2 was all about.

The way I approach these sort of changes is say I tweak the `fanart_data` to now include a new dict key called `keyart`.  I'd then log the full dict `fanart_data` to make sure it has a value and I can then check the url to the artwork to make sure it's as expected.  Now artwork is passed to the skin via `setArt()` and that also has to be changed so your new key/value is passed to the skin in the listbuilder for Movies etc.  Getting the keyart returned from fanarttv inside the dict is ONLY half the battle.

Good learning curve but as you stick with it you'll get better and better.  I'm only 3yrs invested and consider myself mediocre at best...way better python coders out there than I but I still fumble about.  I had a little background in some other languages so for me it was a bit of a cheat.  I'm just no longer into the public scene sticking my neck out to get chopped off..lol 

Yes keyart has no text.  Notice the language here says `00`.  I know I've seen cases of string `None` also and I believe even a singleton of None.  When I once dialed in keyart I used `('00', 'None', None)`
<img width="1085" alt="keyart" src="https://user-images.githubusercontent.com/46383815/176983050-e3940945-c56b-498c-8404-0d1cda689bcf.PNG">
